include/drivers/dai.h

```diff
@@ -0,0 +1,629 @@
+ *         -'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---
+ *              | Word 1            | Word 2            | Word 3  |  Word n |
+ */
+#define DAI_FMT_DATA_FORMAT_PCM_SHORT       (1 << DAI_FMT_DATA_FORMAT_SHIFT)
+
+/* Alias for interface compatibility with SOF and Linux */
+#define DAI_FMT_DATA_FORMAT_PCM_A		DAI_FMT_DATA_FORMAT_PCM_SHORT
+
+/** @brief PCM Long Frame Sync Data Format.
+ *
+ * Serial data is transmitted in two's complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the falling edge
+ * of the clock signal (SCK). The rising edge of the frame sync signal (WS)
+ * indicates the start of the PCM word. The frame sync has an arbitrary length,
+ * however it has to fall before the start of the next frame. An arbitrary
+ * number of data words can be sent in one frame.
+ *
+ *          .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-.
+ *     SCK -' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-
+ *              .--- ---.    ---.        ---.                               .---
+ *     WS      -'       '-      '-          '-                             -'
+ *         -.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---
+ *     SD   |   |MSB|   |...|   |LSB|MSB|   |...|   |LSB|MSB|   |...|   |LSB|
+ *         -'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---
+ *              | Word 1            | Word 2            | Word 3  |  Word n |
+ */
+#define DAI_FMT_DATA_FORMAT_PCM_LONG        (2 << DAI_FMT_DATA_FORMAT_SHIFT)
+
+/* for interface compatibility with SOF and Linux */
+#define DAI_FMT_DATA_FORMAT_PCM_B		DAI_FMT_DATA_FORMAT_PCM_LONG
+
+/**
+ * @brief Left Justified Data Format.
+ *
+ * Serial data is transmitted in two's complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the rising edge
+ * of the clock signal (SCK). The bits within the data word are left justified
+ * such that the MSB is always sent in the clock period following the WS
+ * transition. Left channel data are sent first indicated by WS = 1, followed
+ * by right channel data indicated by WS = 0.
+ *
+ *          .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-.
+ *     SCK -' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-
+ *            .-------------------------------.                               .-
+ *     WS  ---'                               '-------------------------------'
+ *         ---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.-
+ *     SD     |MSB|   |...|   |LSB| x |...| x |MSB|   |...|   |LSB| x |...| x |
+ *         ---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'-
+ *            | Left channel                  | Right channel                 |
+ */
+#define DAI_FMT_DATA_FORMAT_I2S_LEFT_JUSTIFIED  (3 << DAI_FMT_DATA_FORMAT_SHIFT)
```

Comment:
Not sure why I2S (which is just one of many DAIs) details are included here, should rather be in a dedicated I2S header.

Reply:
IMHO very good idea.